phalcon/mvc/model/resultset/simple.zep

```diff
@@ -181,48 +188,80 @@ class Simple extends Resultset
 			 * Get the resultset column map
 			 */
 			let columnMap = this->_columnMap;
-			if typeof columnMap != "array" {
+			if typeof columnMap != "array" && !fireAfterFetch {
 				return records;
 			}
 
-			let renamedRecords = [];
+			let parsedRecords = [];
 			if typeof records == "array" {
 
+				if fireAfterFetch {
+					let modelName = get_class(this->_model);
+					try{
+						let reflectionMethod = new ReflectionMethod(modelName,"afterFetch");
+						let closure = reflectionMethod->getClosure(new {modelName}());
+					}
+					catch ReflectionException, e {
+						let fireAfterFetch = false;
+						if typeof columnMap != "array" {
+							return records;
+						}
+					}
+				}
+
 				for record in records {
 
-					let renamed = [];
-					for key, value in record {
+					if typeof columnMap == "array" {
+						let renamed = [];
+						for key, value in record {
 
-						/**
-						 * Check if the key is part of the column map
-						 */
-						if !fetch renamedKey, columnMap[key] {
-							throw new Exception("Column '" . key . "' is not part of the column map");
-						}
+							/**
+							 * Check if the key is part of the column map
+							 */
+							if !fetch renamedKey, columnMap[key] {
+								throw new Exception("Column '" . key . "' is not part of the column map");
+							}
 
-                        if typeof renamedKey == "array" {
+							if typeof renamedKey == "array" {
 
-		                    if !fetch renamedKey, renamedKey[0] {
-                	            throw new Exception("Column '" . key . "' is not part of the column map");
-                        	}
-						}
+								if !fetch renamedKey, renamedKey[0] {
+									throw new Exception("Column '" . key . "' is not part of the column map");
+								}
+							}
 
-						let renamed[renamedKey] = value;
+							let renamed[renamedKey] = value;
+						}
+						if !fireAfterFetch {
+							let parsedRecords[] = renamed;
+						}
+						else{
+							let parsedRecords[] = this->_executeClosure(closure, renamed);
+						}
+					}
+					else {
+						let parsedRecords[] = this->_executeClosure(closure, record);
 					}
-
-					/**
-					 * Append the renamed records to the main array
-					 */
-					let renamedRecords[] = renamed;
 				}
 			}
 
-			return renamedRecords;
+			return parsedRecords;
 		}
 
 		return records;
 	}
 
+	private function _executeClosure(<Closure> closure, array record) -> array
+	{
+		var rowObject, objectClosure, arrayObject;
+
+		let rowObject = (object)record;
+		let objectClosure = closure->bindTo(rowObject);
+		{objectClosure}();
+		// in zephir casting object to array doesn't work, so we need to use ArrayObject, we could possibly save some time if we could just (array)rowObject
+		let arrayObject = new ArrayObject(rowObject);
```

Comment:
Actually ArrayObject is faster in this case(beacause it's not recursive and we don't need this in this case), maybe zephir is optimizing somehow json_decode ?

http://sandbox.onlinephpfunctions.com/code/6e495c0dffb998052ee3043880f507d38d9aae35